netapp/perf.go

```diff
@@ -0,0 +1,158 @@
+	request := perfObjectGetInstanceRequest{
+		Base: base,
+	}
+	request.Params.XMLName = xml.Name{Local: "perf-object-get-instances"}
+	request.Params.PerfObjectGetInstanceParams = params
+	return &request
+}
+
+type InstanceInfoQuery struct {
+	InstanceInfo *InstanceInfo `xml:"instance-info,omitempty"`
+}
+
+type PerfObjectInstanceListInfoIterParams struct {
+	DesiredAttributes *InstanceInfo      `xml:"desired-attributes,omitempty"`
+	FilterData        string             `xml:"filter-data,omitempty"`
+	MaxRecords        int                `xml:"max-records,omitempty"`
+	ObjectName        string             `xml:"objectname"`
+	Query             *InstanceInfoQuery `xml:"query,omitempty"`
+	Tag               string             `xml:"tag,omitempty"`
+}
+
+type InstanceInfo struct {
+	Name string `xml:"name"`
+	Uuid string `xml:"uuid"`
+}
+
+type PerfObjectInstanceListInfoIterResponse struct {
+	XMLName xml.Name `xml:"netapp"`
+	Results struct {
+		ResultBase
+		AttributesList struct {
+			InstanceInfo []InstanceInfo `xml:"instance-info"`
+		} `xml:"attributes-list"`
+		NextTag    string `xml:"next-tag"`
+		NumRecords int    `xml:"num-records"`
+	} `xml:"results"`
+}
+
+type PerfObjectInstanceListInfoPageResponse struct {
+	Response    *PerfObjectInstanceListInfoIterResponse
+	Error       error
+	RawResponse *http.Response
+}
+
+type perfObjectInstanceListInfoIterRequest struct {
+	Base
+	Params struct {
+		XMLName xml.Name
+		*PerfObjectInstanceListInfoIterParams
+	}
+}
+
+func newPerfObjectInstanceListInfoIterRequest(params *PerfObjectInstanceListInfoIterParams, base Base) *perfObjectInstanceListInfoIterRequest {
+	request := perfObjectInstanceListInfoIterRequest{
+		Base: base,
+	}
+	request.Params.XMLName = xml.Name{Local: "perf-object-instance-list-info-iter"}
+	request.Params.PerfObjectInstanceListInfoIterParams = params
+	return &request
+}
+
+func (p *Perf) PerfObjectGetInstances(params *PerfObjectGetInstanceParams) (*PerfObjectGetInstancesResponse, *http.Response, error) {
+	request := newPerfObjectGetInstanceRequest(params, p.Base)
+	response := PerfObjectGetInstancesResponse{}
+	rawResponse, err := p.get(request, &response)
+	return &response, rawResponse, err
+}
+
+func (p *Perf) PerfObjectInstanceListInfoIter(params *PerfObjectInstanceListInfoIterParams) (*PerfObjectInstanceListInfoIterResponse, *http.Response, error) {
+	request := newPerfObjectInstanceListInfoIterRequest(params, p.Base)
+	response := PerfObjectInstanceListInfoIterResponse{}
+	rawResponse, err := p.get(request, &response)
+	return &response, rawResponse, err
+}
+
+type PerfObjectInstanceListInfoHandler func(PerfObjectInstanceListInfoPageResponse) (shouldContinue bool)
+
+func (p *Perf) PerfObjectInstanceGetAllInfo(options *PerfObjectInstanceListInfoIterParams, fn PerfObjectInstanceListInfoHandler) {
+
+	requestOptions := options
+
+	for shouldContinue := true; shouldContinue; {
```

Comment:
I think that this way of writing is simple
```
for  {
[snip]
  if nexttag != "" && handlerResponse {
    break
  }
}
```

Reply:
you have me at a loss, im not sure what you mean here.

Reply:
I'm telling you that you should not need `shouldContinue`.

Reply:
Trying to keep the codebase consistent - This was patterned after:
https://github.com/pepabo/go-netapp/blob/master/netapp/aggregate.go#L52
https://github.com/pepabo/go-netapp/pull/5/files/aa590074789adebcf3e970ef9bded90fb89dfb3b#diff-942f1ae318e034f883df819be23f8641

Reply:
I understand. Thank you for keeping the policy.
I will fix it by myself.